docs/ruleset_functions/get_energy_required_to_heat_swh_use.md

```diff
@@ -0,0 +1,91 @@
+## get_energy_required_to_heat_swh_use
+
+Description: This function calculates the total energy required to heat the SWH use over the course of a year in btu.  Note - this function does not work for service water heating uses with use_units == "OTHER".  In this case, it will return 0.
+
+Inputs:
+- **swh_use**
+- **RMD**
+- **building_segment**
+
+Returns:
+- **energy_required_by_space**: A dict where the keys are space_ids and values are the total energy required to heat the swh_use for that space.  If a swh_use is not assigned to any spaces, the key will be "NO_SPACES_ASSIGNED"
+
+Function Call:
+
+- get_spaces_served_by_SWH_use
+- get_obj_by_id
+
+Data Lookup: None
+
+Logic:
+
+- get the schedule id: `hourly_schedule_id = swh_use.use_multiplier_schedule`
+- get the distribution system id: `distribution_system_id = swh_use.served_by_distribution_system`
+- get the temperature at fixture: `supply_temperature = distribution_system.design_supply_temperature`
+- get the water used: `swh_use_value = swh_use.use`
+- get the inlet water temperature schedule id: `inlet_temperature_schedule_id = distribution_system.entering_water_mains_temperature_schedule`
+- set the drain heat recovery efficiency to 0: `drain_heat_recovery_efficiency = 0`
+- get the drain heat recovery efficiency if heat is recovered by drain: `if(swh_use.is_heat_recovered_by_drain): drain_heat_recovery_efficiency = distribution_system.drain_heat_recovery_efficiency`
+- get the use_units: `use_units = swh_use.use_units`
+
+- now we need to get all of the spaces that the swh_use is applied to.  The convention is that if any spaces reference the swh_use, then the service water heating use applies to only those spaces. If no spaces reference the service water heating use, it applies to all spaces in the building segment.
+- get the ids of spaces served by this swh_use: `space_ids = get_spaces_served_by_SWH_use(RMD, swh_use)`
+- convert the space ids to spaces: `spaces = [get_obj_by_id(space_id, RMD) for space_id in space_ids]`
+- if there are no spaces, and the use_units are anything other than POWER and VOLUME, the spaces list should include all spaces in the building segment: `if len(spaces) == 0 and use_units not in ["POWER","VOLUME"]: spaces = building_segment.spaces`
```

Comment:
Is `building_segment` a part of the same `RMD`? I did not understand the use of `building_segment` here. Could you clarify? @weilixu @KarenWGard @JacksonJ-KC